sdk/eventhub/Azure.Messaging.EventHubs/tests/Primitives/EventProcessorTests.cs

```diff
@@ -3,11 +3,13 @@
 
 using System;
 using System.Collections.Generic;
+using System.Linq;
```

Comment:
Its starting to feel like we should consider splitting this into multiple classes, but I'm not sure how to segment them.   Would be interested in thoughts on this.

Reply:
I think that doing something similar to what you did in the [Event Processor Base Proposal](https://gist.github.com/jsquire/1f4a1e72fc02871f443ce365222d40f6#azuremessagingeventhubsprimitives) would be good (constructors, optional extension points, core required extension points, storage required extension points). Not ideal, but I think it could work if we moved stuff around (like putting all the "handlers" together). We would need a class to test internal / private stuff.

I cannot guarantee the distribution would be good, though. I think it's hard to estimate how many tests each file would have.

Reply:
That's a really good thought.   I wonder if we could just break them up with partial classes nested under the same file, so its still one logical unit but physically broken down into categories.    I'll probably defer that until after this closes, but definitely worth looking into.

Reply:
I tried breaking down into partial classes by responsibility.  That worked just fine, technically, but there was enough overlap with shared functionality that made it difficult to follow what was where and how it was being used, especially because file nesting wasn't playing nice.  For now, I think I'm going to leave them as-is and revisit once implementation is complete.